### DISCUSSION TITLE
discount on items not part of xml file

### DISCUSSION THREAD

**Discussion by fisfro** [stephanstapel/ZUGFeRD-csharp#461] [Q&A]
Hello again,

If I have discounts on each item, I will get the expected validation error: 
Invoice line net amount MUST equal (Invoiced quantity * (Item net price/item price base quantity) + Sum of invoice line charge amount - sum of invoice line allowance amount
Pfad: /ubl:Invoice/cac:InvoiceLine[1]

because partprice * quantity will be not the position price due the fact that the discount will be not considered.

therefore I have changed the tradeLineItem configuration as following:
            TradeLineItem tradeLineItem = new TradeLineItem(pos.Pos.ToString())
            {
                // lineID = pos.Pos.ToString(),
                SellerAssignedID = pos.ArtikelNummer,
                Name = pos.Bezeichnung.UnmaskHTMLinString(),
                Description = pos.Beschreibung.UnmaskHTMLinString(),
                UnitCode = GetQuantityCodes(pos.Einheit), // Mengeneinheit
                BilledQuantity = pos.Menge,
                NetUnitPrice = pos.PPGesamt / pos.Menge,//pos.EP,
                LineTotalAmount = pos.PPGesamt,
                TaxPercent = pos.Steuersatz,
                TaxType = TaxTypes.VAT,
                TaxCategoryCode = GetTaxCategoryCodes(pos.SteuertypID),
                BillingPeriodStart = Beleg.LeistungszeitraumBeginn,
                BillingPeriodEnd = Beleg.LeistungszeitraumEnde,
            };

            // Rabatte zuweisen, falls vorhanden
            if (pos.RabattGesamt > 0)
            {
                tradeLineItem.AddSpecifiedTradeAllowanceCharge(
                    isDiscount: true, // Rabatt
                    currency: CurrencyCodes.EUR,
                    basisAmount: pos.PPGesamt, // Der Basisbetrag, auf den sich der Rabatt bezieht
                    actualAmount: pos.RabattGesamt, // Rabattbetrag
                    reason: "Rabatt für Position " + pos.Pos // Optional: Grund für den Rabatt
                );
            }
            // TradeLineItem der Beschreibung hinzufügen
            desc.TradeLineItems.Add(tradeLineItem);

so checking the tradeLineItem during adding to list includes the TradeAllowanceCharge, but unfourtunately this information will be not part of the xml-file. 

If I change item price to reduced item price and the I will pass the validation but at the end ,the discount information is missing in the invoice, which seems to be a problem- I think.

Do I something wrong or is this part not supported?

Thanks in advance!

Robert

**Comment by hschilke** [comment]
I had the same problem.
For my first tests I yoused the profile XRechnung.
Try to use another profile:
desc.Save(stream, ZUGFeRDVersion.Version23, Profile.Extended);

Result is:
        <ram:SpecifiedTradeAllowanceCharge>
          <ram:ChargeIndicator>
            <udt:Indicator>false</udt:Indicator>
          </ram:ChargeIndicator>
          <ram:CalculationPercent>25.00</ram:CalculationPercent>
          **<ram:BasisAmount>2000.00</ram:BasisAmount>**
          <ram:ActualAmount>1500.00</ram:ActualAmount>
          <ram:Reason>Rabatt</ram:Reason>
        </ram:SpecifiedTradeAllowanceCharge>

**Comment by stephanstapel** [comment]
yes, you are right. That's a bug in the library
